PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azuretools/ijidea/utility/AzureAnAction.java

```diff
@@ -88,6 +88,28 @@ public void sendTelemetryOnAction(AnActionEvent anActionEvent, final String acti
             null, buildProp(anActionEvent, extraInfo));
     }
 
+    public void sendTelemetryOnSuccess(AnActionEvent anActionEvent, Map<String, String> extraInfo) {
+        sendTelemetryOnAction(anActionEvent, "Success", extraInfo);
+    }
+
+    public void sendTelemetryOnException(AnActionEvent anActionEvent, Throwable e) {
+        Map<String, String> extraInfo = new HashMap<>();
+        extraInfo.put("ErrorMessage", e.getMessage());
+        this.sendTelemetryOnAction(anActionEvent, "Exception", extraInfo);
+    }
+
+    protected String getServiceName(AnActionEvent event) {
+        return TelemetryConstants.ACTION;
+    }
+
+    protected String getOperationName(AnActionEvent event) {
+        try {
+            return event.getPresentation().getText().toLowerCase().trim().replace(" ", "-");
+        } catch (Exception ignore) {
+            return "";
+        }
```

Comment:
replace "", "" with StringUtils ?

Reply:
it is an legacy code, I just move the method sequence